test/Intlc/LinterSpec.hs

```diff
@@ -0,0 +1,30 @@
+module Intlc.LinterSpec where
+
+import           Intlc.ICU
+import           Intlc.Linter
+import           Prelude
+import           Test.Hspec
+
+spec :: Spec
+spec = describe "linter" $ do
+  it "lints static text" $ do
+    lint (Static "Hello world") `shouldBe` Success
+
+  it "lints dynamic streams with 1 plain text token" $ do
+    lint (Dynamic (fromList [Plaintext "yay"])) `shouldBe` Success
+
+  it "lints dynamic streams with 2 or more plain text token" $ do
+    lint (Dynamic (fromList [Plaintext "yay", Plaintext "Hello"])) `shouldBe` Success
+
+  it "lints dynamic streams with 1 dynamic simple interpolation" $ do
+    lint (Dynamic (fromList [Interpolation (Arg "Hello" String)])) `shouldBe` Success
+
+  it "lints dynamic streams with 1 dynamic complex interpolation" $ do
+    lint (Dynamic (fromList [Interpolation (Arg "Hello" (Callback []))])) `shouldBe` Success
+
+  it "lints dynamic streams with 1 dynamic complex interpolation and 1 simple interpolation" $ do
+    lint (Dynamic (fromList [Interpolation (Arg "Hello" (Callback [])), Plaintext "hello"])) `shouldBe` Success
+
+  it "does not lint dynamic streams with 2 or more dynamic complex interpolations" $ do
+    lint (Dynamic (fromList [Interpolation (Arg "Hello" (Callback [])), Interpolation (Arg "Hello" (Bool [] []))])) `shouldBe` Failure "Too many interpolations. They will appear nested once flattened."
```

Comment:
Is there a way to clean these up? e.g property based tests maybe?
Also is there any way to assert on `Failure` with any strings so we don't have to fix this test if we update the error message?

Reply:
Raised an issue for PBTs: https://github.com/unsplash/intlc/issues/124

On the error checking, does `Failure` need to take `Text`? Could we change the type to only denote all the failure states (currently just one), then handle rendering that outside of `lint`?

Reply:
Cool call. Let me see what I can do

Reply:
74869cd6aee42bfdf060a48a877b8f3e301d3be1